sklearn/linear_model/ransac.py

```diff
@@ -395,12 +418,36 @@ def fit(self, X, y, sample_weight=None):
 
         # if none of the iterations met the required criteria
         if inlier_mask_best is None:
-            raise ValueError(
-                "RANSAC could not find valid consensus set, because"
-                " either the `residual_threshold` rejected all the samples or"
-                " `is_data_valid` and `is_model_valid` returned False for all"
-                " `max_trials` randomly ""chosen sub-samples. Consider "
-                "relaxing the ""constraints.")
+            if ((self.n_skips_no_inliers_ + self.n_skips_invalid_data_ +
+                    self.n_skips_invalid_model_) > self.max_skips):
+                raise ValueError(
+                    "RANSAC skipped more iterations than `max_skips` without"
+                    " finding a valid consensus set. Iterations were skipped"
+                    " because each randomly chosen sub-sample failed the"
+                    " passing criteria. The object attributes"
```

Comment:
object -> estimator. But perhaps "See estimator attributes for diagnostics (n_skips*)." is sufficient